stonesoup/feeder/tests/test_geo.py

```diff
@@ -43,15 +45,76 @@ def test_utm_converter(detector):
 
     p_east = float('-inf')
     assert converter.zone_number is None
+    assert converter.zone_letter is None
 
-    for long, (time, detections) in zip(range(-3, 4), converter):
+    for long, (time, detections) in enumerate(converter, -3):
         detection = detections.pop()
+        lat = 50 + long
 
         assert converter.zone_number == 30
-        assert converter.northern
+        assert converter.zone_letter == 'T'
+        assert detection.metadata['utm_zone'] == (30, 'T')
 
         assert p_east < detection.state_vector[0]
         p_east = detection.state_vector[0]
 
-        assert pytest.approx((50, long), rel=1e-2, abs=1e-4) == utm.to_latlon(
+        assert pytest.approx((lat, long), rel=1e-2, abs=1e-4) == utm.to_latlon(
             *detection.state_vector[0:2], zone_number=30, northern=True)
+
+
+def test_utm_set_zone(detector):
+    # Note, zone letter doesn't matter unless it's north vs. south of the equator
+    converter = LongLatToUTMConverter(detector, zone_number=31, zone_letter='U')
+
+    p_east = float('-inf')
+    assert converter.zone_number == 31
+    assert converter.zone_letter == 'U'
+
+    for long, (time, detections) in enumerate(converter, -3):
+        detection = detections.pop()
+        lat = 50 + long
+
+        assert converter.zone_number == 31
+        assert converter.zone_letter == 'U'
+        assert detection.metadata['utm_zone'] == (31, 'U')
+
+        assert p_east < detection.state_vector[0]
+        p_east = detection.state_vector[0]
+
+        assert pytest.approx((lat, long), rel=1e-2, abs=1e-4) == utm.to_latlon(
+            *detection.state_vector[0:2], zone_number=31, zone_letter='U', strict=False)
+
+        # Correct zone shouldn't match, as conversion is wrong
+        assert pytest.approx((lat, long)) != utm.to_latlon(
+            *detection.state_vector[0:2], zone_number=30, zone_letter='U', strict=False)
+
+
+def test_utm_converter_dynamic(detector):
```

Comment:
Adding a test for the case when `dynamic`, `zone_number` and `zone_letter` attributes are all pre-set, raising `ValueError("Cannot have both dynamic and fixed zone") ` will increase code coverage to 100%